aten/src/ATen/native/BatchLinearAlgebraKernel.cpp

```diff
@@ -1092,9 +1102,13 @@ void unpack_pivots_cpu_kernel(TensorIterator& iter, const int64_t dim_size) {
       const auto pivots_data = reinterpret_cast<const int32_t*>(pivots_ptr);
 
       for (const auto i : c10::irange(dim_size)) {
+        auto new_idx = pivots_data[i] - 1;
+        TORCH_CHECK(new_idx >= 0 && new_idx < max_pivot,
+                    "pivots passed to lu_unpack must be between 1 and LU.size(-1)."
+                    "Did you properly pass the result of lu_factor?");
```

Comment:
`LU.size(-1)` works in the context of `lu_solve`, but not in the context of generic `lu`. The LU implementation swaps rows, not columns.

Reply:
Makes sense. I will update both to keep them consistent.